piksi_tools/interpolate_event_positions.py

```diff
@@ -0,0 +1,146 @@
+"""
+interpolate_event_trigger.py allows user to analyze log files
+with external event triggers.
+"""
+
+from sbp.client.loggers.json_logger import JSONLogIterator
+from sbp.ext_events import SBP_MSG_EXT_EVENT
+import csv
+
+def lin_interp(o, n, otow, ntow, ttow):
+  """
+  Linearly interpolates distance.
+  
+  Parameters
+  ----------
+  o : integer
+    distance previous to trigger
+  n : integer
+    distance prior to trigger
+  otow : integer
+    TOW of data packet previous to trigger
+  ntow : integer
+    TOW of data packet after trigger
+  ttow : integer
+    TOW of trigger 
+  """
+  d = float(n-o)
+  t = (ntow-otow) 
+  v = d/t
+  return o+(v*(ttow-otow))
+
+
+def write_positions(infile, outfile, msgtype):
+  """
+  Organize and output data to log file.
+
+  Parameters
+  ----------
+  infile : string
+    Log file to get data from.
+  outfile : string
+    Output file.
+  """
+
+  with JSONLogIterator(infile) as log:
+    log = log.next()
+    lasttow = 0
+    deltalasttrigger = 0
+    positiondesplayed = 1
+    debouncetime = 1000
+    previous_msg = None
+    fileoutput = open(outfile, 'wt')
+    writer = csv.writer(fileoutput)
+    if msgtype == 'MsgBaselineNED' :
+      writer.writerow(("TOW (ms)","N (mm)","E (mm)","D (mm)","# of Sats",
+                     "Flags","H Accuracy (mm)","V Accuracy (mm)"))
+    if msgtype == 'MsgPosECEF' :
+      writer.writerow(("TOW (ms)","X (m)","Y (m)","Z (m)","# of Sats",
+                     "Flags","Accuracy (mm)"))
+    if msgtype == 'MsgPosLLH' :
+      writer.writerow(("TOW (ms)","Lat (deg)","Lon (deg)","Height (m)","# of Sats",
+                     "Flags","H Accuracy (mm)","V Accuracy (mm)"))
+    if msgtype == 'MsgBaselineECEF' :
+      writer.writerow(("TOW (ms)","X (mm)","Y (mm)","Z (mm)","# of Sats",
+                     "Flags","Accuracy (mm)")) 
+
+    while True:
+      try:
+        msg, metadata = log.next()
+        hostdelta = metadata['delta']
+        hosttimestamp = metadata['timestamp']
+        
+        if msg.__class__.__name__ == "MsgExtEvent" :
+          deltalasttrigger = msg.tow - lasttow
+          if deltalasttrigger > debouncetime:
+            lasttow = msg.tow
+            positiondesplayed = 0
+
+        if msg.__class__.__name__ == msgtype and positiondesplayed == 0 :
```

Comment:
each time through the loop the msg has only one type, so you should be able to make these all "else if" statements.  This will make it clearer to the next coder how the control flow works.